### DISCUSSION TITLE
Register pinia plugin with @pinia/nuxt module?

### DISCUSSION THREAD

**Discussion by ttkoma** [vuejs/pinia#622] [Help and Questions] [answered]
Hey! Thanks for your work, pinia a very cool.

Please any idea how  i can register pinia plugin with @pinia/nuxt module?

**Comment by posva** [comment]
It's done for you. Is something missing at https://pinia.esm.dev/ssr/nuxt.html?

**Comment by ttkoma** [comment, marked as answer]
Thanks for this module. But I didn't find an example in the documentation on how to use the Pinia plugins to extend the store using nuxt.
It turned out to be very simple, we can use nuxt plugins to access the pinia instance.
like this:
#### **`nuxt.config.js`**
``` js
plugins: [
    "./plugins/secretPiniaPlugin.js"
 ],
```

#### **`secretPiniaPlugin.js`**
``` js
function SecretPiniaPlugin(ctx) {
  ctx.store.$subscribe((mutation) => {
    // react to store changes
    console.log(`[STORE] ${mutation.storeId} ${mutation.type} changed`)
  })

  return { secret: 'the cake is a lie' }
}

export default function (nuxtCtx) {
  const { pinia } = nuxtCtx;
  pinia.use((piniaCtx) => SecretPiniaPlugin(piniaCtx));
}

```

**Reply by posva** [reply]
aah, I thought you were talking about the Vue plugin of Pinia 😆 

Yeah, this is the correct way. I will document it

**Reply by posva** [reply]
Full instructions at https://pinia.esm.dev/core-concepts/plugins.html#nuxt-js
